.github/workflows/embedding-sdk.yml

```diff
@@ -91,6 +91,35 @@ jobs:
       - name: Run frontend embedding SDK snippets type check
         run: yarn run embedding-sdk:docs-snippets:type-check
 
+  embedding-sdk-no-exposed-internal-types-validate:
+    needs: [ files-changed, build ]
+    if: |
+      needs.files-changed.outputs.frontend_embedding_sdk_sources == 'true' &&
+      needs.build.result == 'success'
```

Comment:
We run it on any FE code change.

It means that for other teams the CI may fail if they leak internal types from types that are exposed from SDK.
In practice it almost should not happen, we re-export a few `plugin` and `question` types from `frontend/src/metabase/embedding-sdk/types/*` but it is our namespace